### ISSUE TITLE
CNI: netdev conflict when moving to pod namespace

### ISSUE THREAD

**Issue by lmilleri** [ovn-org/ovn-kubernetes#3867]
Sometimes ovn-k CNI is not able to move the netdev from the host to the pod namespace. This happens mainly for SR-IOV/vDPA interfaces when enabling the ovn-k secondary interface.

This is the faulty scenario:
- netdev in the host to be moved to pod namespace is called `eth0`
- the pod has already an interface called `eth0` (primary interface)
- when the user creates a new pod (planning to use the above interface), the pod creation fails with the message:

`I0815 09:13:00.351681    4692 cni.go:280] [vdpa/vdpa-pod2 45fe5372d3625d82795c00442454c7332331d571efb14f6440ec1f8bded678d7 network l2-network NAD vdpa/ovn-kubernetes-b] ADD finished CNI request [vdpa/vdpa-pod2 45fe5372d3625d82795c00442454c7332331d571efb14f6440ec1f8bded678d7 network l2-network NAD vdpa/ovn-kubernetes-b], result "", err failed to configure pod interface: failed to move device eth0 to netns: "file exists"`

So basically, `eth0` can't be moved to pod namespace because conflicting with the existing `eth0` interface.

I think a possible solution to this problem is to rename the interface before moving it to the pod.
